utils/jsonSigning.js

```diff
@@ -0,0 +1,74 @@
+// JSON Cleartext Signing
+'use strict'
+
+const rsasign = require('jsrsasign') // eslint-disable-line no-unused-vars
+const ServerError = require('../errors/server-error')
+const _ = require('lodash')
+
+// Sign JSON object, using ECDSA P-256 private key and public key.
+// Insert 'signature' block in the JSON object.
+// Warning: there must not be an existing 'signature' block.
+module.exports.sign = function (json, prvKey, pubKey) {
+  if (!prvKey) throw new ServerError('Problem reading private key for JSON signing')
+  if (!pubKey) throw new ServerError('Problem reading public key for JSON signing')
+  const strJWS = KJUR.jws.JWS.sign('ES256', // eslint-disable-line no-undef
+                                   JSON.stringify({alg: 'ES256'}),
+                                   JSON.stringify(json),
+                                   prvKey)
+  const jwsArray = strJWS.split('.')
+  // Get x and y coordinates
+  const pubKeyHex = pubKey.pubKeyHex
+  // According to jsrsasign documentation,
+  // pubkey hex length should be 2 (header) + 64 (X) + 64 (Y) = 130.
+  // first two chracters must be "04".
+  if (pubKeyHex.length !== 130 || !_.startsWith(pubKeyHex, '04')) {
+    throw new ServerError('Problem decoding public key for JSON signing')
+  }
+  const hX = pubKeyHex.slice(2, 66)
+  const hY = pubKeyHex.slice(66, 130)
+
+  const signedJSON = _.cloneDeep(json)
+  signedJSON.signature = {
+    'algorithm': 'ES256',
+    'publicKey': {
+      'type': 'EC',
+      'curve': 'P-256',
+      'x': hX,
+      'y': hY
+    },
+    'value': jwsArray[2]
+  }
+  return signedJSON
+}
+
+// Verify 'signature' block on the input JSON object has the correct signature value.
+module.exports.verify = function (json, pubKey) {
+  if (!json || !json.signature || !json.signature.value) {
+    throw new ServerError('Invalid input for JSON verification')
```

Comment:
We don't really have granular error handling in five-bells-shared. This error would likely be due to a poor implementation in a client module (the connector calls `verify` with malformed json, for example), in which case we need a generic catch-all 500 error. The connector should check the json before passing to this method, and throw an appropriate HTTP error, or wrap calls to `verify` in a `try/catch` block, to re-throw the error as an HTTP 4xx. The two alternatives I see here are
1. throw an `Error`, but we decided to only throw typed error in RILP-330.
2. create a new type of error and throw that instead